sdmetrics/reports/single_table/quality_report.py

```diff
@@ -0,0 +1,100 @@
+"""Single table quality report."""
+
+import itertools
+import sys
+
+import numpy as np
+import pandas as pd
+import tqdm
+
+from sdmetrics.single_table import (
+    ContingencySimilarity, CorrelationSimilarity, KSComplement, TVComplement)
+
+
+class QualityReport():
+    """Single table quality report."""
+
+    METRICS = {
+        'Column Shapes': [KSComplement, TVComplement],
+        'Column Pair Trends': [CorrelationSimilarity, ContingencySimilarity],
+    }
+
+    _metric_results = {}
+
+    def generate(self, real_data, synthetic_data, metadata):
+        """Generate report.
+
+        Args:
+            real_data (pandas.DataFrame):
+                The real data.
+            synthetic_data (pandas.DataFrame):
+                The synthetic data.
+            metadata (dict):
+                The metadata, which contains each column's data type as well as relationships.
+        """
+        metrics = list(itertools.chain.from_iterable(self.METRICS.values()))
+
+        for metric in tqdm.tqdm(metrics, desc='Creating report:'):
+            self._metric_results[metric.__name__] = metric.compute_breakdown(
+                real_data, synthetic_data)
+
+        self._property_breakdown = {}
```

Comment:
can we add an `__init__` and define this there? Reasons for it being explained here https://softwareengineering.stackexchange.com/questions/357937/is-it-really-correct-to-declare-all-instance-attributes-in-init